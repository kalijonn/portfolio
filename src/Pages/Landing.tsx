import React from 'react'
import {useNavigate} from 'react-router-dom'
import kalijonnImage from './kalijonn.png'

export default function Landing() {
  const navigate = useNavigate()
  return (
    <div className="flex h-1/2">
      <div className="font-bodoni w-1/2 flex flex-col justify-center">
        <h1 className="font-bold text-5xl my-10 leading-none">
          <span className="text-cyan-500">Hi! I'm Kali.</span> I'm passionate
          about all things Typescript.
        </h1>
        <p className="font-sans">And maybe Javascript.</p>
        <div>
          <button
            className="py-2 px-3 my-5 mr-5 bg-cyan-500 text-white text-sm font-semibold rounded-md shadow-lg shadow-cyan-500/50 focus:outline-none font-sans"
            onClick={() => {
              navigate('/contact')
            }}
          >
            Contact Me!
          </button>
          <button className="py-2 px-3 my-5 text-sm font-semibold rounded-md font-sans underline">
            Learn More
          </button>
        </div>
      </div>
      <div>
        <img
          src={kalijonnImage}
          alt="Kali's headshot"
          className="object-cover rounded-lg"
        />
      </div>
    </div>
  )
}
