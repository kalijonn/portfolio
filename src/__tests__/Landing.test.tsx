import React from 'react'
import {render} from '@testing-library/react'
import {axe} from 'jest-axe'
import Landing from 'Pages/Landing'
import {MemoryRouter} from 'react-router-dom'

test('landing no violations', async () => {
  const {container} = render(
    <MemoryRouter>
      <Landing />
    </MemoryRouter>,
  )
  const results = await axe(container)
  expect(results).toHaveNoViolations()
})

test('landing to have correct text', () => {
  const {getByRole} = render(
    <MemoryRouter>
      <Landing />
    </MemoryRouter>,
  )
  expect(getByRole('heading')).toHaveTextContent(
    "Hi! I'm Kali. I'm passionate about all things Typescript.",
  )
})
