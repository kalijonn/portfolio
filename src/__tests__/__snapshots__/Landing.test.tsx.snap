// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing no violations 1`] = `
<div>
  <div
    class="flex h-1/2"
  >
    <div
      class="font-bodoni w-1/2 flex flex-col justify-center"
    >
      <h1
        class="font-bold text-5xl my-10 leading-none"
      >
        <span
          class="text-cyan-500"
        >
          Hi! I'm Kali.
        </span>
         I'm passionate about all things Typescript.
      </h1>
      <p
        class="font-sans"
      >
        And maybe Javascript.
      </p>
      <div>
        <button
          class="py-2 px-3 my-5 mr-5 bg-cyan-500 text-white text-sm font-semibold rounded-md shadow-lg shadow-cyan-500/50 focus:outline-none font-sans"
        >
          Contact Me!
        </button>
        <button
          class="py-2 px-3 my-5 text-sm font-semibold rounded-md font-sans underline"
        >
          Learn More
        </button>
      </div>
    </div>
    <div>
      <img
        alt="Kali's headshot"
        class="object-cover rounded-lg"
        src=""
      />
    </div>
  </div>
</div>
`;
